packages/jest-cli/src/__tests__/run_jest_with_coverage.test.js

```diff
@@ -0,0 +1,109 @@
+import runJest from '../run_jest';
+
+jest.mock('jest-util');
+
+jest.mock(
+  '../test_scheduler',
+  () =>
+    class {
+      constructor(globalConfig) {
+        this._globalConfig = globalConfig;
+      }
+
+      scheduleTests() {
+        return {_globalConfig: this._globalConfig};
+      }
+    },
+);
+
+jest.mock(
+  '../test_sequencer',
+  () =>
+    class {
+      sort(allTests) {
+        return allTests;
+      }
+      cacheResults() {}
+    },
+);
+
+jest.mock(
+  '../search_source',
+  () =>
+    class {
+      constructor(context) {
+        this._context = context;
+      }
+
+      async getTestPaths(globalConfig, changedFilesPromise) {
+        const {files} = await changedFilesPromise;
+        const paths = files.filter(path => path.match(/__tests__/));
+
+        return {
+          collectCoverageFrom: files.filter(path => !path.match(/__tests__/)),
+          tests: paths.map(path => ({
+            context: this._context,
+            duration: null,
+            path,
+          })),
+        };
+      }
+    },
+);
+
+const config = {roots: [], testPathIgnorePatterns: [], testRegex: ''};
+let globalConfig;
+const defaults = {
+  changedFilesPromise: Promise.resolve({
+    files: ['foo.js', '__tests__/foo-test.js', 'dont/cover.js'],
+  }),
+  contexts: [{config}],
+  onComplete: runResults => (globalConfig = runResults._globalConfig),
+  outputStream: {},
+  startRun: {},
+  testWatcher: {isInterrupted: () => false},
+};
+
+describe('collectCoverageFrom patterns', () => {
+  it('should apply collectCoverageFrom patterns coming from SearchSource', async () => {
+    expect.assertions(1);
+
+    await runJest({
+      ...defaults,
```

Comment:
We don't transpile object spread, please use `Object.assign`

Reply:
Ah yeah, I pushed the changes so should be good now 😄